boards/x86/up_squared/up_squared_defconfig

```diff
@@ -13,3 +13,7 @@ CONFIG_SMP=y
 CONFIG_MP_NUM_CPUS=2
 CONFIG_BUILD_OUTPUT_EFI=y
 CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_APIC_TSC_DEADLINE_TIMER=y
+CONFIG_HPET_TIMER=n
+CONFIG_APIC_TIMER=n
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=1593600000
```

Comment:
FWIW: This rate was hand-measured on my board using some hackery in the HPET driver.  I'm currently trying to turn that into a rig we can commit to samples somewhere that will detect and calibrate HPET, TSC and APIC clocks so we don't have to constantly guess at this stuff.  Will add to this PR if there's time, otherwise submit separately.

Reply:
Ok, so it is choice between using APIC_TIMER and APIC_TSC_DEADLINE_TIMER. I did not realize that at first. Something to think about, maybe we could simplify this need for 3 configs to choose a timer.

Reply:
I'm trying to test this driver with acrn_ehl_crb and see some regressions in tests/kernel/timer. I added the same lines as this up2 defconfig, except using HW cycles for ACRN. 1 of 4 completes  tests/kernel/timer/timer_error_case/kernel.timer, the other 3 configurations hang:
- tests/kernel/timer/timer_monotonic/kernel.timer.monotonic - test_monotomic_timer
- tests/kernel/timer/timer_api/kernel.timer - test_timer_duration_period
- tests/kernel/timer/timer_api/kernel.timer.tickless - test_timer_duration_period

Reply:
Trying the same exercise on ehl_crb (which typically uses HPET configuration), fails differently than acrn:
2 pass and 2 fail. 
- tests/kernel/timer/timer_api/kernel.timer - regression on test_timer_remaining, and hangs on test_timeout_abs
- tests/kernel/timer/timer_api/kernel.timer.tickless - regression on test_timer_remaining, and hangs on test_timeout_abs

```
START - test_timer_remaining
Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/timer/timer_api/src/main.c:718: test_timer_remaining: (abs(delta_ticks) <= MAX(slew_ticks, 1U) is false)
tick/busy slew -2 larger than test threshold 0
FAIL - test_timer_remaining in 1.978 seconds
===================================================================
START - test_timeout_abs
```
(hangs)

Reply:
So CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC needs to be the actual TSC counter rate for the hardware, not the value copied from up_squared (though my guess is that is probably close) and absolutely not a value left over from an HPET driver, which is 100x too low.  The 1.9 GHz number you have there certainly sounds about right.

Reply:
Sure, I understand. I was not reusing the up2 value. What seems to be the issue is that config is required with the new driver?

Reply:
**Update:** All tests under kernel are passing except kernel/common/bootdelay with the below changes in defconfig.
CONFIG_APIC_TSC_DEADLINE_TIMER=y
CONFIG_HPET_TIMER=n
CONFIG_APIC_TIMER=n
CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=1900000000

Reply:
This now works great on acrn_ehl_crb (spot tested with twister the tests/kernel/timer/ and tests/kernel/common/.